lib/interactive_viewer.ml

```diff
@@ -111,6 +104,65 @@ let change_summary z_patches : ui Lwd.t =
   in
   W.string ~attr:Notty.A.(fg lightcyan) operation_count
 
+(** Side by side diff view implementation **)
+
+type view_mode = SideBySide | Normal
+
+let view_mode = Lwd.var Normal
+
+let toggle_view_mode () =
+  match Lwd.peek view_mode with
+  | Normal -> Lwd.set view_mode SideBySide
+  | SideBySide -> Lwd.set view_mode Normal
+
+let rec split_and_align_hunk hunks mine_acc their_acc =
+  match hunks with
+  | [] -> (List.rev mine_acc, List.rev their_acc)
+  | (`Common _ as common) :: t ->
+      split_and_align_hunk t (common :: mine_acc) (common :: their_acc)
+  | `Mine s :: `Their s' :: t ->
+      split_and_align_hunk t (`Mine s :: mine_acc) (`Their s' :: their_acc)
```

Comment:
This pairs only one added line with one removed line. This might not be the output we expect, for example:
```
18 11     W.scrollbox
19 12       (W.vbox
20 13          [
21    -          pure_str "Hello world!";
22    -          string_of_counter counter_d;
   14 +          pure_str (string_of_operation patch.operation);
23 15            Lwd.pure
24 16            @@ Ui.keyboard_area
```

This diff renders like this in side-by-side:
```
  W.scrollbox                                                 W.scrollbox
    (W.vbox                                                     (W.vbox
       [                                                           [
         pure_str "Hello world!";
         string_of_counter counter_d;                                pure_str (string_of_operation patch.operation);
         Lwd.pure                                                    Lwd.pure
         @@ Ui.keyboard_area                                         @@ Ui.keyboard_area
```

It would be more intuitive if the whole block of removed line was paired with the added line.
Visually aligning the first removed line of the block with the first added line, leaving the empty line in the right side to be below the green line instead of above.